### PR TITLE
Fix problem report url for cc

### PR DESCRIPTION
**Pull request by reinsch82**
The doc url supplied to the TAPI was incorrect.

Fixed this by using url and not just `toString()`

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by reinsch82** [comment]
@bot-gradle test this

File: platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/internal/cc/impl/ConfigurationCacheProblemsServiceIntegTest.groovy [hunk 1]
```diff
@@ -69,7 +69,14 @@ class ConfigurationCacheProblemsServiceIntegTest extends AbstractConfigurationCa
             fqid == REGISTRATION_UNSUPPORTED
             contextualLabel == "registration of listener on 'Gradle.buildFinished' is unsupported"
             definition.severity == Severity.WARNING
-            definition.documentationLink != null
+            definition.documentationLink.url.endsWith("/userguide/configuration_cache.html#config_cache:requirements:build_listeners")
+            locations.size() == 2
```
Comment (hegyibalint) [inline platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/internal/cc/impl/ConfigurationCacheProblemsServiceIntegTest.groovy:73]: :x:  We should stop doing this, as implicit locations (e.g., task location, which I'm trying to introduce) break this.

File: platforms/core-configuration/configuration-cache/src/crossVersionTest/groovy/org/gradle/internal/cc/impl/ConfigurationCacheProblemsCrossVersionTest.groovy [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+        @Override
+        void statusChanged(ProgressEvent event) {
+            if (event instanceof SingleProblemEvent) {
+                this.problems.add(event)
+            }
+        }
+    }
+
+    def "Failing executions produce problems"() {
```
Comment (hegyibalint) [inline platforms/core-configuration/configuration-cache/src/crossVersionTest/groovy/org/gradle/internal/cc/impl/ConfigurationCacheProblemsCrossVersionTest.groovy:47]: 🤔 Most of our tests are starting with lowercase, are TAPI tests different?

```suggestion
    def "failing executions produce problems"() {
```

File: platforms/core-configuration/configuration-cache/src/crossVersionTest/groovy/org/gradle/internal/cc/impl/ConfigurationCacheProblemsCrossVersionTest.groovy [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+        then:
+        thrown(BuildException)
+        listener.problems.size() == 1
+        verifyAll(listener.problems[0]) {
+            definition.id.displayName == "registration of listener on 'Gradle.buildFinished' is unsupported"
+            definition.id.group.displayName == "configuration cache validation"
+            definition.id.group.name == "configuration-cache"
+            definition.severity == Severity.ERROR
+            locations.size() == 2
```
Comment (hegyibalint) [inline platforms/core-configuration/configuration-cache/src/crossVersionTest/groovy/org/gradle/internal/cc/impl/ConfigurationCacheProblemsCrossVersionTest.groovy:76]: ❌ We should stop doing this, as implicit locations (e.g., task location, which I'm trying to introduce) break this.

**Review by hegyibalint — CHANGES_REQUESTED** [review]
Implementation is OK, left some remarks about location counting

**Review by hegyibalint — APPROVED** [review]

**Pull request merged**